caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -716,10 +716,20 @@ func (st *ServerType) serversFromPairings(
 					}
 				}
 
+				// If TLS is specified as directive, it will also result in 1 or more connection policy being created
+				// Thus, catch-all address with non-standard port, e.g. :8443, can have TLS enabled without
+				// specifying prefix "https://"
+				// Second part of the condition is to allow creating TLS conn policy even though `auto_https` has been disabled
+				// ensuring compatibility with behavior described in below link
+				// https://caddy.community/t/making-sense-of-auto-https-and-why-disabling-it-still-serves-https-instead-of-http/9761
+				createdTLSConnPolicies, ok := sblock.pile["tls.connection_policy"]
+				hasTLSDirectiveSpecified := (ok && len(createdTLSConnPolicies) > 0) ||
```

Comment:
I think naming the variable like this would be more correct, because TLS can be enabled by things other than the TLS directive, as the condition implies.

```suggestion
				hasTLSEnabled := (ok && len(createdTLSConnPolicies) > 0) ||
```

Reply:
I've renamed the variabled to make it more variable. PTAL. Thanks!